### ISSUE TITLE
Creation of 'FunctionSpace(mesh, ("DG", 0))' fails on some platforms

### ISSUE THREAD

**Issue by mrambausek** [FEniCS/dolfinx#959]
The following code fails with SIGABRT on Fedora 31 with gcc 9.3.1 and scotch built via petsc 3.12.3 from source (it does not fail in the CI tests and also not in a fairly recent dolfinx container)
```
from mpi4py import MPI
from dolfinx import UnitSquareMesh, FunctionSpace
mesh = UnitSquareMesh(MPI.COMM_WORLD, 1, 1)
V = FunctionSpace(mesh, ("DG", 0))
```
The actual crash happens in repartioning via SCOTCH called from `compute_reordering_map`
in `DofMapBuilder.cpp`. For the case of "DG 0", an empty graph (`AdjacencyList`) is sent to `SCOTCH::compute_gps` because all diagonal entries are removed from the initial graph an nothing remains for "DG 0". 
I have no clue, why it fails on Fedora and not on Ubuntu.

**Comment by mrambausek** [comment]
This is printed out 
```
/usr/include/c++/9/bits/stl_vector.h:1042: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](std::vector<_Tp, _Alloc>::size_type) [with _Tp = int; _Alloc = std::allocator<int>; std::vector<_Tp, _Alloc>::reference = int&; std::vector<_Tp, _Alloc>::size_type = long unsigned int]: Assertion '__builtin_expect(__n < this->size(), true)' failed.
```

**Comment by chrisrichardson** [comment]
Some builds of SCOTCH are prone to crashing on empty graphs.

**Comment by mrambausek** [comment]
Actually, it turned out to be not SCOTCH but an assertion in std::vector::operator[] (see PR #961)

**Comment by garth-wells** [comment]
Fixed in #961.
